### PR TITLE
Use dataprotection API

### PR DESCRIPTION
**Pull request by volllly**
- [x] Store Launcher Config Encrypted
- [x] Use windows DPAPI
- [x] Use linux Keyring
- [x] Implement fallback
- [x] Store Cashbox Config Encrypted

### CODE REVIEW

**Comment by volllly** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by volllly** [comment]
recheck

File: src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+namespace fiskaltrust.Launcher.Extensions
+{
+
+    unsafe static partial class KeyUtils
+    {
+        private const Int64 KEYCTL = 250;
+
+        [LibraryImport("libc", SetLastError = true, EntryPoint = "syscall")]
+        private static partial Int64 keyctl_read(Int64 syscall, Int32 cmd, Int32 key, byte* buffer, UIntPtr buflen);
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs:20]: Any specific reason for not using the aliases for the respective datatypes in this class? (`int`, `long`, ..)

Comment (volllly) [reply src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs:20]: It helped me not to get confused with the similarly named but not necessary equal C datatypes on the other side ^^ for these things where the bit size of the type is more important than the value range I like to use those 🤷

File: src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+using fiskaltrust.Launcher.Common.Helpers;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.AspNetCore.DataProtection.XmlEncryption;
+using Serilog;
+
+namespace fiskaltrust.Launcher.Extensions
+{
+
+    unsafe static partial class KeyUtils
```
[the remark this answers is not in the document]
Comment (volllly) [reply src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs:15]: The `LibraryImport` attribute needs it to be partial ^^

[the remark this answers is not in the document]
Comment (volllly) [reply src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs:15]: I will link the manpages 👌

File: src/fiskaltrust.Launcher/Commands/ConfigCommand.cs [hunk 3]
```diff
@@ -69,9 +67,11 @@ public async Task<int> InvokeAsync(InvocationContext context)
                     return 1;
                 }
 
+                dataProtector = DataProtectionExtensions.Create(CipherAccessToken ?? launcherConfiguration?.AccessToken).CreateProtector("fiskaltrust.Launcher.Configuration");
```
[the remark this answers is not in the document]
Comment (volllly) [reply src/fiskaltrust.Launcher/Commands/ConfigCommand.cs]: I've provided error messages for both cases 👌

File: src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs:15]: image

[the remark this answers is not in the document]
Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Extensions/DataProtectionExtensions.cs:15]: homer

**Review by TSchmiedlechner — DISMISSED** [review]

**Review by TSchmiedlechner — APPROVED** [review]

**Pull request merged**